### PR TITLE
Refactor/best and worst offense

### PR DESCRIPTION
**Pull request by jgthomas-12**
Refactored and added the worst method. Made 3 helper methods and tests to go alongside them. Just committing and pushing before I need to start a new branch and feature.

After that I sent it to turning main and lost this in the ether for like 15 minutes. Just found it. Still  haven't quelled my fear of this insipid site.

### CODE REVIEW

**Review by WZale — APPROVED** [review]
Helper methods continue to look good, as does your work on retrieving things from the ether.

File: lib/stat_tracker.rb [hunk 1]
```diff
@@ -90,41 +90,26 @@ def count_of_teams
...
+      value
+    end
+    worst_name = @teams.select do |team|
+      return team.teamname if worst[0] == team.team_id
     end
-    best_average
   end
 
+  #helper method returns hash home/away team_id and total goals scored
   def total_goals
```
Comment (Jredish11) [inline lib/stat_tracker.rb:113]: nice these will come in handy!

File: lib/stat_tracker.rb [hunk 2]
```diff
@@ -142,4 +128,17 @@ def total_games
     team_games
   end
 
+  #helper method returns hash home/away team_id and average goals scored
+  def averages_id_by_goals_games
+    average = Hash.new(0)
+    total_goals.each do |id, goals|
+      total_games.each do |eye_d, games|
+        if id == eye_d
```
Comment (Jredish11) [inline lib/stat_tracker.rb:136]: eye eye captain

**Review by Jredish11 — APPROVED** [review]

File: lib/stat_tracker.rb [hunk 3]
```diff
@@ -134,6 +119,7 @@ def total_goals
     team_goals
   end
 
+  #helper method returns hash home/away team_id and total games played
   def total_games
```
Comment (Jredish11) [inline lib/stat_tracker.rb:123]: these will come in handy. also potential for a module refactor?. something to think about

**Pull request merged**